Monika After Story/game/definitions.rpy

```diff
@@ -5,6 +5,102 @@ define config.developer = False #This is the flag for Developer tools
+
+        IN:
+            field_name - the reference to the object
+            args - not sure, but renpy doesn't use it (passes in an empty tuple)
+            kwargs - the store modules where renpy will look for the object
+
+        OUT:
+            tuple of the object and its key
+        """
+        # if it's a function, we eval it
+        if "(" in field_name:
+            # id to split the string
+            split_index = field_name.index("(")
+            # get the func's name
+            func_name = field_name[:split_index]
+            # get the arg and kwargs
+            args = field_name[split_index:]
+
+            # it still may include store modules, get the 1st if there's any
+            if "." in func_name:
+                func_store_name, func_name = func_name.split(".", 1)
+                first = func_store_name
+                func_store_name += "."
+
+            else:
+                func_store_name = ""
+                first = func_name
+
+            # now we find the store's name to use in eval
+            # iterate through all the scopes you passed in to find the one
+            # where the function/module was defined
+            if isinstance(kwargs, renpy.substitutions.MultipleDict):
+                for scope in kwargs.dicts:
+                    if first in scope:
+                        stores_names_list = [
+                            store_obj_name
+                            for store_obj_name, store_obj in sys.modules.iteritems()
+                            if store_obj and store_obj.__dict__ is scope
+                        ]
+                        if stores_names_list:
+                            scope_store_name = "renpy.{0}.".format(stores_names_list[0])
+                            break
+
+                # if we got to this, you passed in something undefined
+                else:
+                    scope_store_name = ""
+
+            # if you've not passed anything, use the base store
+            elif first in kwargs:
+                scope_store_name = "renpy.store."
+
+            # if we got to this, you passed in something undefined
+            else:
+                scope_store_name = ""
+
+            # finally get the value from the function
+            obj = eval("{0}{1}{2}{3}".format(scope_store_name, func_store_name, func_name, args))
+
+        # otherwise just get the reference
+        else:
+            first, rest = field_name._formatter_field_name_split()
+
+            obj = self.get_value(first, args, kwargs)
+
+            for is_attr, i in rest:
+                if is_attr:
+                    obj = getattr(obj, i)
+
+                else:
+                    if not isinstance(i, long):
+                        i = long(i)
+                    obj = obj[i]
+
+        return obj, first
+
+    # allows us to use a more advanced string formatting
+    renpy.substitutions.Formatter.get_field = get_field_override
```

Comment:
so looking at where this lies, `Formatter` is an extension of the `string.Formatter` class, which is a python class. Rather than overwrite the function (which in reality is just adding a definition to renpy's `Formatter` so it doesn't fallback to python's formatter), I think you should just extend `Formatter`, define `get_field` on your extended class, and set `renpy.substitutions.formatter` equal to an instance of your `Formatter` extended class.